tests/standalone_tests/test_cleanup_after_algorithm_execution.py

```diff
@@ -318,92 +315,94 @@ def cleaner(controller_config, node_landscape_aggregator):
         contextids_cleanup_folder=controller_config.cleanup.contextids_cleanup_folder,
         node_landscape_aggregator=node_landscape_aggregator,
     )
+    return Cleaner(cleaner_init_params)
 
-    cleaner = Cleaner(cleaner_init_params)
-
-    # Cleaner class is defined as singleton. So there is always a single object.
-    # Test function test_cleanup_after_uninterrupted_algorithm_execution_triggered_by_timelimit
-    # alters the release timelimit, so it is reset here.
-    # TODO Cleaner class maybe should not be implemented as singleton??
-    controller_config = AttrDict(controller_config)
-    cleaner._contextid_release_timelimit = (
-        controller_config.cleanup.contextid_release_timelimit
-    )
 
-    return cleaner
+@pytest.fixture
+def db_cursors(
+    monetdb_globalnode,
+    monetdb_localnode1,
+    monetdb_localnode2,
+    monetdb_localnodetmp,
+    globalnode_db_cursor,
+    localnode1_db_cursor,
+    localnode2_db_cursor,
+    localnodetmp_db_cursor,
+):
+    # TODO the node_ids should not be hardcoded..
+    return {
+        "testglobalnode": globalnode_db_cursor,
+        "testlocalnode1": localnode1_db_cursor,
+        "testlocalnode2": localnode2_db_cursor,
+        "testlocalnodetmp": localnodetmp_db_cursor,
+    }
 
 
 @pytest.mark.slow
 @pytest.mark.very_slow
-@pytest.mark.asyncio
-async def test_cleanup_after_uninterrupted_algorithm_execution(
+def test_synchronous_cleanup(
     context_id,
-    algorithm,
-    engine,
     cleaner,
     node_landscape_aggregator,
-    controller,
     reset_celery_app_factory,  # celery tasks fail if this is not reset
+    db_cursors,
 ):
 
     # Cleaner gets info about the nodes via the NodeLandscapeAggregator
-    # poll NodeLandscapeAggregator until it has some node info
+    # Poll NodeLandscapeAggregator until it has some node info
     start = time.time()
     while (
         not node_landscape_aggregator.get_nodes()
         or not node_landscape_aggregator.get_cdes_per_data_model()
         or not node_landscape_aggregator.get_datasets_locations()
     ):
+
         if time.time() - start > NLA_WAIT_TIME_LIMIT:
             pytest.fail(
-                "Exceeded max retries while waiting for the node landscape aggregator "
+                "Exceeded max retries while waiting for the node landscape aggregator to"
                 "return some nodes"
             )
         time.sleep(0.5)
```

Comment:
If you call the nla._update instead of waiting for the nla to asynchronously update the registries, you can lower the time that each test needs.
Or extract those lines in a function  because they are repeated 5 times exactly the same.

Reply:
this would not work because the NodeLandscapeAggregator is also used by the Cleaner so it has to be updated in there as well. The correct solution for these test would be to become unit tests mocking all external system dependencies.

Reply:
What do you mean `it has to be updated in there as well`?
NLA is a singleton, if you update it here instead of waiting, nla will be updated for the rest of the components(cleaner included).
In the while loop you are using the nla that the cleaner is using anyway.
Am I missing something?